packages/liveblocks-react-ui/src/primitives/Collapsible/index.tsx

```diff
@@ -9,39 +9,58 @@ import {
   useRef,
 } from "react";
 
+import { useControllableState } from "../../utils/use-controllable-state";
 import type { ContentProps, RootProps, TriggerProps } from "./types";
 
-/* -------------------------------------------------------------------------------------------------
- * Root
- * -----------------------------------------------------------------------------------------------*/
-
-const RootContext = createContext<{
-  open: boolean;
-  onOpenChange(open: boolean): void;
+const COLLAPSIBLE_ROOT_NAME = "CollapsibleRoot";
+const COLLAPSIBLE_TRIGGER_NAME = "CollapsibleTrigger";
+const COLLAPSIBLE_CONTENT_NAME = "CollapsibleContent";
 
+const CollapsibleContext = createContext<{
+  open?: boolean;
+  onOpenChange: (open: boolean) => void;
   disabled: boolean;
   contentId: string;
 } | null>(null);
 
-export const Root = forwardRef<HTMLDivElement, RootProps>(
+/* -------------------------------------------------------------------------------------------------
+ * Root
+ * -----------------------------------------------------------------------------------------------*/
+
+const CollapsibleRoot = forwardRef<HTMLDivElement, RootProps>(
   (
-    { open, onOpenChange, disabled = false, asChild, ...props },
+    {
+      open: controlledOpen,
+      onOpenChange: controlledOnOpenChange,
+      defaultOpen,
+      disabled = false,
+      asChild,
+      ...props
+    },
     forwardedRef
   ) => {
+    const [isOpen, onOpenChange] = useControllableState(
+      // If the collapsible is neither controlled nor uncontrolled, it defaults to controlled as open.
+      controlledOpen === undefined && defaultOpen === undefined
+        ? true
+        : controlledOpen,
+      controlledOnOpenChange,
+      defaultOpen
+    );
```

Comment:
I'm okay to keep us but I try to not complicate the component logic with more things than needed. In this case, I'm okay with keeping the component simply uncontrolled because it makes the internal logic much simpler and the primitives as dumb as possible. I know `useControllableState` handles a lot of the complexity but it also adds an extra layer of code to manage and account for.

Reply:
Would you say the same for a public API? It's still private for now but the goal is to have it public at some point, and even now I had in mind to add `defaultOpen`/`open`/`onOpenChange` to `AiTool` which is collapsible and maybe already public. imo requiring your own `useState` for a public API like this would be a bit much, supporting controlled/uncontrolled seems to be an expectation for stateful components (this one but also `Switch`, `Checkbox`, `Input`, etc) in "behavior" libraries like Radix.

Reply:
For public API, I agree it makes sense to provide both controlled and uncontrolled props. If the goal is to make these public or use it directly on a public API, please feel free to ignore my comments! Thanks for explaining @marcbouchenoire.